### PR TITLE
xorg.xkeyboardconfig_custom: update for 2.39

### LINKED ISSUE #243837: XKB extraLayouts: Makefile.am: No such file or directory
### Describe the bug
I use the following listing in `/etc/nixos/configuration.nix` to install a custom keyboard layout:

```nix
services.xserver.extraLayouts.james = {
  description = "Personal layout";
  languages = [ "eng" ];
  symbolsFile = xkb/symbols/james;
  keycodesFile = xkb/keycodes/james;
};
```

This refers to files in the following structure:

```console
[jamesw@nixos:/etc/nixos]$ tree xkb/
xkb/
├── keycodes
│   └── james
├── map.xkb
├── rules
│   ├── base
│   └── base.xml.in
└── symbols
    ├── extend
    ├── james
    ├── level3
    ├── level5
    └── us

3 directories, 9 files
```

This worked until recently (last updated a few months ago), but now fails to build:

<details>
<summary>Full trace</summary>
```console
[jamesw@nixos:/etc/nixos]$ sudo nixos-rebuild build -I nixpkgs=/home/jamesw/nixpkgs
[sudo] password for jamesw: 
building Nix...
building the system configuration...
these 18 derivations will be built:
  /nix/store/bl23arq8z3gx6zbnpxf8khiimnb8qkzj-xkeyboard-config-2.39.drv
  /nix/store/ajl23r5jmg6gwscrvk7x5bmw0bn7qgha-xkb-console-keymap.drv
  /nix/store/7r3g741as20czs7va4ljxb1y1ahkndgv-vconsole.conf.drv
  /nix/store/6pk592bsnviqaj0qvzjbvsy2j17lpvbw-X-Restart-Triggers.drv
  /nix/store/azrh9ksz7pxlnxl0020w5brddza8gzrm-set-environment.drv
  /nix/store/bwpg4zqnajcjq2dc8cgvi9hpll9bbgnr-xkb-validated.drv
  /nix/store/c98zg1pdk2lfs8cvgs324rsgx8r2gnrc-etc-environment.drv
  /nix/store/d7h68dlz31qc7k6kmhpjc1pnngk55msj-etc-profile.drv
  /nix/store/nc7c3crd00mgm8z1zi03l10zcv5fgw79-unit-reload-systemd-vconsole-setup.service.drv
  /nix/store/lzms0gpripar2h5z4izidfghqwb2zcb5-system-units.drv
  /nix/store/w8vd7qw3c67x19zm2pvmss573jr7k6sr-xserver-wrapper.drv
  /nix/store/z7bqn1zmnpkdra8kcz6fa67f0ncghys4-sddm.conf.drv
  /nix/store/hxayn5rq3zpx61h9rs4h39z0d9j27ks9-etc.drv
  /nix/store/q5xh0c9aqlfr2i79c1088fwywm7zfwfz-keymap.drv
  /nix/store/n28vq50k63s7r49ls2gx3jzw406s6c30-stage-1-init.sh.drv
  /nix/store/wa7y591fh49ykm0h7pjmjavm6n9kyvra-initrd-linux-6.1.38.drv
  /nix/store/l2wqw7pl34jb6wglqjhz2s3vgqz4wmig-boot.json.drv
  /nix/store/rhza22pchhmszlpn92kghy5i7zc54gjx-nixos-system-nixos-23.11.git.8acef304efe7.drv
building '/nix/store/bl23arq8z3gx6zbnpxf8khiimnb8qkzj-xkeyboard-config-2.39.drv'...
unpacking sources
unpacking source archive /nix/store/1vzcm443ijpwd2y3byv3xb4rf3nraypr-xkeyboard-config-2.39.tar.xz
source root is xkeyboard-config-2.39
setting SOURCE_DATE_EPOCH to timestamp 1686270106 of file xkeyboard-config-2.39/types/pc
patching sources
patching script interpreter paths in rules/merge.py rules/compat/map-variants.py rules/xml2lst.pl
rules/merge.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/9c03r86hcdn43dm3hsgjirifvyzfkhwh-python3-3.10.12/bin/python3"
rules/compat/map-variants.py: interpreter directive changed from "#!/usr/bin/env python3" to "/nix/store/9c03r86hcdn43dm3hsgjirifvyzfkhwh-python3-3.10.12/bin/python3"
rules/xml2lst.pl: interpreter directive changed from "#!/usr/bin/env perl" to "/nix/store/wmccghnpfng3hmpgks56k62izma8bvhy-perl-5.36.0/bin/perl"
keycodes/Makefile.am: No such file or directory
?
No match
8
/nix/store/fzb9wy1yz0hn69vxw12954szvrjnjjgk-stdenv-linux/setup: line 144: pop_var_context: head of shell_variables not a function context
error: builder for '/nix/store/bl23arq8z3gx6zbnpxf8khiimnb8qkzj-xkeyboard-config-2.39.drv' failed with exit code 1
error: 1 dependencies of derivation '/nix/store/c98zg1pdk2lfs8cvgs324rsgx8r2gnrc-etc-environment.drv' failed to build
error: 1 dependencies of derivation '/nix/store/azrh9ksz7pxlnxl0020w5brddza8gzrm-set-environment.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ajl23r5jmg6gwscrvk7x5bmw0bn7qgha-xkb-console-keymap.drv' failed to build
error: 1 dependencies of derivation '/nix/store/bwpg4zqnajcjq2dc8cgvi9hpll9bbgnr-xkb-validated.drv' failed to build
error: 1 dependencies of derivation '/nix/store/w8vd7qw3c67x19zm2pvmss573jr7k6sr-xserver-wrapper.drv' failed to build
error: 1 dependencies of derivation '/nix/store/d7h68dlz31qc7k6kmhpjc1pnngk55msj-etc-profile.drv' failed to build
error: 1 dependencies of derivation '/nix/store/q5xh0c9aqlfr2i79c1088fwywm7zfwfz-keymap.drv' failed to build
error: 1 dependencies of derivation '/nix/store/z7bqn1zmnpkdra8kcz6fa67f0ncghys4-sddm.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/7r3g741as20czs7va4ljxb1y1ahkndgv-vconsole.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/6pk592bsnviqaj0qvzjbvsy2j17lpvbw-X-Restart-Triggers.drv' failed to build
error: 1 dependencies of derivation '/nix/store/n28vq50k63s7r49ls2gx3jzw406s6c30-stage-1-init.sh.drv' failed to build
error: 1 dependencies of derivation '/nix/store/wa7y591fh49ykm0h7pjmjavm6n9kyvra-initrd-linux-6.1.38.drv' failed to build
error: 1 dependencies of derivation '/nix/store/nc7c3crd00mgm8z1zi03l10zcv5fgw79-unit-reload-systemd-vconsole-setup.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/l2wqw7pl34jb6wglqjhz2s3vgqz4wmig-boot.json.drv' failed to build
error: 1 dependencies of derivation '/nix/store/lzms0gpripar2h5z4izidfghqwb2zcb5-system-units.drv' failed to build
error: 6 dependencies of derivation '/nix/store/hxayn5rq3zpx61h9rs4h39z0d9j27ks9-etc.drv' failed to build
error: 4 dependencies of derivation '/nix/store/rhza22pchhmszlpn92kghy5i7zc54gjx-nixos-system-nixos-23.11.git.8acef304efe7.drv' failed to build
```
</details>

```console
keycodes/Makefile.am: No such file or directory
?
No match
8
/nix/store/fzb9wy1yz0hn69vxw12954szvrjnjjgk-stdenv-linux/setup: line 144: pop_var_context: head of shell_variables not a function context
```

Commenting out the `keycodesFile` line causes the same error but for `symbols/Makefile.am`. Adding in plausible `Makefile.am` files in the suggested places also doesn't fix the issue.

### Steps To Reproduce
1. Set up the configuration as above.
2. Run `nixos-rebuild build`.

### Expected behavior
The system successfully builds, and I have access to the custom layout.

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[jamesw@nixos:/etc/nixos]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.28, NixOS, 23.05 (Stoat), 23.05.git.17a689596b7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"home-manager, nixos"`
 - channels(jamesw): `""`
 - nixpkgs: `/home/jamesw/nixpkgs`
```

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Description of changes

This change fixes `services.xserver.extraLayouts` on unstable.
Closes #243837.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- Tested
  - [x] custom symbols file (`nixosTests.keymap.custom`)
  - [ ] test other types
- [x] Tested compilation of all packages that depend on this change (none)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rnhmjoj** [comment]
@grahamcOfBorg test keymap.custom

File: pkgs/servers/x11/xorg/overrides.nix [hunk 1]
```diff
@@ -579,28 +579,6 @@ self: super:
...
-        a
-        ${name} \\
-        .
-        w
-        EOF
-        done
-
         # add model description
         ${ed}/bin/ed -v rules/base.xml <<EOF
         /<\/modelList>
```
Comment (ncfavier) [inline pkgs/servers/x11/xorg/overrides.nix:584]: Why do we add a model, btw?

**Comment by nagy** [comment]
This fixes the issue for me, thank you.

File: pkgs/servers/x11/xorg/overrides.nix (hunk 1 above)
Comment (rnhmjoj) [reply pkgs/servers/x11/xorg/overrides.nix:584]: I did a few years ago, IIRC without this the layout is not selectable from a DE, possibly not even from setxkbmap.

**Comment by laMudri** [comment]
Fixes it for me too, with symbols and keycodes!

**Comment by rnhmjoj** [comment]
Thank you for testing!

**Comment by ncfavier** [comment]
Have you tried rebooting, or at least logging out of X and back in?

**Comment by rski** [comment]
yeah, although I think this is my fault; I'm running the x server from my nixos config and setxkbmap from the home-manager config. I had not switched nixos to follow the same version of nixpkgs as home-manager, so it was using the old xorg. 
Similarly, on my other laptop ubuntu provides xorg, so it's a different version than setxkbmap. I can fix it on my side, and that's probably the only way if it's a mismatch between xorg and setxkbmap

**Pull request merged**